src/System.Collections.Specialized/tests/NameObjectCollectionBase/MyNameObjectCollection.cs

```diff
@@ -10,7 +10,13 @@ public class MyNameObjectCollection : NameObjectCollectionBase
         public MyNameObjectCollection(int capacity) : base(capacity) { }
         public MyNameObjectCollection(IEqualityComparer comparer) : base(comparer) { }
         public MyNameObjectCollection(int capacity, IEqualityComparer comparer) : base(capacity, comparer) { }
-        
+#if netstandard17
+#pragma warning disable 618
+        public MyNameObjectCollection(IHashCodeProvider hashProvider, IComparer comparer) : base(hashProvider, comparer) { }
+        public MyNameObjectCollection(int capacity, IHashCodeProvider hashProvider, IComparer comparer) : base(capacity, hashProvider, comparer) { }
+#pragma warning restore 618
+#endif //netstandard17
+
```

Comment:
Yes partial classes in new files is the preferred way. We should only fallback to #ifdef's if there isn't any other clean way to split the code up.